services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -223,8 +223,6 @@ public void run()
             )
         );
         injector.getInstance(ExecutorLifecycle.class).join();
-        // Explicitly call lifecycle stop, dont rely on shutdown hook.
-        lifecycle.stop();
```

Comment:
I think peon process and other druid processes have different contract when it comes to stopping. 

For the non-peon druid process, expectation is that they will `always` run and will stop only when a signal is sent and lifecycle.stop() is indeed called via the shutdown hook. (all non-peon druid process call https://github.com/metamx/java-util/blob/master/src/main/java/com/metamx/common/lifecycle/Lifecycle.java#L283 )
For the peon process however, there are 2 ways for it to stop, one is by sending the signal and another is that it finishes its work and lifecycle.stop() can/should be called in both cases.

BTW, I wouldn't want to hold the PR if people are, in general, agreeing with the approach chosen by this PR :) . I would be happy to merge this if there is agreement on this. Let us bring this up in the syncup.